students/carlos_novoa/lesson04/mailroom.py

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env python3
+import os
+import sys
+import datetime
+
+"""
+Lesson4 - Mailroom, Part 2
+"""
+
+now = datetime.datetime.now()
+
+_donors = [
+    [
+        {
+            'first_name': 'Jim',
+            'last_name': 'Halpert'
+        },
+        {
+            'donations': [
+                1.00
+            ]
+        }
+    ],
+    [
+        {
+            'first_name': 'Pam',
+            'last_name': 'Beesley'
+        },
+        {
+            'donations': [
+                1000.00,
+                2000.00,
+                3000.00
+            ]
+        }
+    ],
+    [
+        {
+            'first_name': 'Dwight',
+            'last_name': 'Shrute'
+        },
+        {
+            'donations': [
+                2.00,
+                3.00
+            ]
+        }
+    ],
+    [
+        {
+            'first_name': 'Michael',
+            'last_name': 'Scott'
+        },
+        {
+            'donations': [
+                10.00,
+                20.00,
+                30.00
+            ]
+        }
+    ],
+    [
+        {
+            'first_name': 'Andy',
+            'last_name': 'Bernard'
+        },
+        {
+            'donations': [
+                500.00
+            ]
+        }
+    ]
+]
+
+
+def submit_donor():
+    donor_name = input("Please enter donor name: ")
+
+    # request donation
+    donation = request_donation()
+
+    # donor lookup
+    donor_found = donor_lookup(donor_name)
+
+    if donor_found:
+        # append existing donor
+        append_donation(donor_name, donation)
+    else:
+        # add new donor
+        add_donor(donor_name, donation)
+
+    donor_email(donor_name, donation)
+    exit_menu()
+    return
+
+
+def letters():
+    cwd = os.getcwd()
+    date = now.strftime('%Y-%m-%d')
+    path = cwd + '/letters/'
+    ext = '.txt'
+    for d in _donors:
+        fn = d[0]['first_name']
+        ln = d[0]['last_name']
+        file_path = "{}{}_{}_{}{}".format(path, date, fn, ln, ext)
+        donations = format_donation(sum(d[1]['donations']))
+        with open(file_path, 'w') as file:
+            text = ('Dear {} {},\n\n'
+                    '        Thank you for your very kind donation of ${}.\n\n'
+                    '        It will be put to very good use.\n\n'
+                    '               Sincerely,\n'
+                    '                  -The Team')
+            body = text.format(fn, ln, donations)
+            file.write(body)
+    return
+
+
+def donor_name_split(donor_name):
+    donor_name = donor_name.split(' ')
+    return [donor_name[0], donor_name[1]]
```

Comment:
this is recreating what `donor_name.split(' ')` returns which means you don't need a function for this and can do it inline

this is what i mean when i say they are equal
```
donor_name.split() == [donor_name[0], donor_name[1]]
```